src/generic-container.test.ts

```diff
@@ -19,6 +22,39 @@ describe("GenericContainer", () => {
     await expect(fetch(url)).rejects.toThrowError();
   });
 
+  it("should be assigned to network and reachable by name", async () => {
+
+    const dockerode = new Dockerode();
```

Comment:
We've extracted a `DockerClient` interface to avoid dealing with `Dockerode` directly, as it's sometimes too low level. It also uses callbacks whereas `testcontainers` aims to use Promises and async/await.

Please implement the required functionality in the `DockerodeDockerClient` class.